experimental/CliffordAlgebras/src/CliffordAlgebras.jl

```diff
@@ -0,0 +1,584 @@
+so it is returned as a vector containing the basis elements. The first one is the
+multiplicative identity of $C$. The square of the second basis element, if present,
+equals `quadratic_discriminant(C)`.
+"""
+function centroid(C::CliffordAlgebra)
+  if isdefined(C, :centroid)
+    return C.centroid
+  end
+  n = _dim_qf(C)
+  if n == 0
+    C.centroid = [one(C)]
+    return C.centroid
+  end
+  T = orthogonal_basis(space(C))
+  orth_elt = prod(map(i -> sum(map(j -> gen(C, j) * T[i, j], 1:n)), 1:n))
+  orth_elt *= denominator(orth_elt)
+  C.disq = coefficients(orth_elt^2)[1]
+  C.centroid = [one(C), orth_elt]
+end
+
+@doc raw"""
+    quadratic_discriminant(C::CliffordAlgebra) -> FieldElem
+
+Return the quadratic discriminant of $C$ as an element of `base_ring(C)`.
+"""
+function quadratic_discriminant(C::CliffordAlgebra)
+  if isdefined(C, :disq)
+    return C.disq
+  end
+  if dim(space(C)) == 0
+    C.disq = one(base_ring(C))
+  else
+    C.disq = coefficients(centroid(C)[2]^2)[1]
+  end
+end
+
+@doc raw"""
+    disq(C::CliffordAlgebra) -> FieldElem
+
+Alias for `quadratic_discriminant`.
+"""
+disq(C::CliffordAlgebra) = quadratic_discriminant(C)
+
+@doc raw"""
+    center(C::CliffordAlgebra) -> Vector{CliffordAlgebraElem}
+
+Return the center of $C$. It equals `centroid(C)`, if and only if `dim(space(C))`
+is odd. Otherwise it is one-dimensional and returned as a tuple of length one
+containing the multiplicative identity of $C$.
+"""
+function center(C::CliffordAlgebra)
+  if isdefined(C, :center)
+    return C.center
+  end
+  if is_odd(dim(space(C)))
+    C.center = centroid(C)
+  else
+    C.center = [one(C)]
+  end
+end
+
+@doc raw"""
+    representation_matrix(x::CliffordAlgebraElem) -> MatElem
+
+Return the representation matrix of the element $x$ with respect to `basis(parent(x))`.
+"""
+function representation_matrix(x::CliffordAlgebraElem, action::Symbol=:left)
+  @req (action == :left) || (action == :right) "The action is either $(:left) or $(:right)."
+  C = parent(x)
+  n = dim(C)
+  res = zero_matrix(base_ring(C), n, n)
+  if action == :left
+    for i in 1:n
+      res[:, i] = coefficients(x * basis(C, i))
+    end
+  elseif action == :right
+    for i in 1:n
+      res[:, i] = coefficients(basis(C, i) * x)
+    end
+  end
+  return res
+end
+
+################################################################################
+#
+#  unary operators
+#
+################################################################################
+
+
+Base.:-(x::CliffordAlgebraElem) = parent(x)(map(y -> -1 * y, coefficients(x)))
+
+################################################################################
+#
+#  binary operators
+#
+################################################################################
+
+function Base.:+(x::CliffordAlgebraElem{T}, y::CliffordAlgebraElem{T}) where {T<:FieldElem}
+  @req parent(x) === parent(y) "The inputs must lie in the same Clifford algebra"
```

Comment:
```suggestion
  check_parent(x, y)
```
Same everywhere else